Implement Overkiz async_execute_commands

nyroDev opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
While implementing a new entity in PR #78659, I noticed multiple commands was always executing one by one.
This is not usaly a problem for simple entity because they call one command at once.
But for climate entities in the Atlantic/Somfy world, they usually need multiple commands to work correctly.
Doing only 1 call with multiple commands instead of multiple call for each command is great because:
- it only counts at one api CALL for rate limits
- it's done this way through the API
- the canceling could be done for all commands at once
- it only create 1 history log on overkiz side

The current entity that will benefit this change for now are:
- [x] somfy_thermostat (3 or 4 calls down to 1)
- [ ] atlantic_heat_recovery_ventilation (3 or 4 calls down to 1)

I will do the changes on other entity when design of the function is correct and somfy_thermostat is validated


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

nyroDev committed:
Implement async_execute_commands and use it somfy_thermostat

homeassistant commented:
Hey there @imicknl, @vlebourl, @tetienne, mind taking a look at this pull request as it has been labeled with an integration (`overkiz`) you are listed as a [code owner](https://github.com/home-assistant/core/blob/dev/CODEOWNERS#L825) for? Thanks!

nyroDev commented:
Marked as draft as it's waiting further discussion and implementing details may need to change
previous comment on other PR
https://github.com/home-assistant/core/pull/78659#pullrequestreview-1129128907
https://github.com/home-assistant/core/pull/78659#pullrequestreview-1124827510

tetienne commented on homeassistant/components/overkiz/executor.py:
```diff
@@ -73,24 +73,64 @@ def select_attribute(self, *attributes: str) -> OverkizStateType:
+        commands: list[list | str] = [[command_name, parameters]]
+        await self.async_execute_commands(commands)
+
+    # commands except an array of commands
```
```suggestion
    # commands accept an array of commands
```

nyroDev committed:
Update homeassistant/components/overkiz/executor.py

Co-authored-by: Thibaut <thibaut@etienne.pw>

nyroDev committed:
Migrate AtlanticPassAPCHeatingZone to multiple commands

MislavMandaric committed:
Allow setting hvac mode through set_temperature climate method in Gree integration (#101196)

* Allow setting hvac mode through set_temperature climate method

* Suggested code simplification when reading hvac mode

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Remove unnecessary temperature unit handling from set temperature with hvac mode tests

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

emontnemery committed:
Fix google_tasks todo tests (#103098)

emontnemery committed:
Fix shopping_list todo tests (#103100)

emontnemery committed:
Fix todoist todo tests (#103101)

emontnemery committed:
Fix local_todo todo tests (#103099)

gjohansson-ST committed:
Bump pytrafikverket to 0.3.8 (#103080)

emontnemery committed:
Add todo to core files (#103102)

manzanotti committed:
Update geniushub-client to v0.7.1 (#103071)

gjohansson-ST committed:
Fix Met Device Info (#103082)

engrbm87 committed:
Code cleanup for transmission integration (#103078)

DCSBL committed:
Bumb python-homewizard-energy to 3.1.0 (#103011)

Co-authored-by: Franck Nijhof <git@frenck.dev>

gjohansson-ST committed:
Add serial to Sensibo (#103089)

emontnemery committed:
Fix restore state for light when saved attribute is None (#103096)

bramkragten committed:
Update frontend to 20231030.0 (#103086)

edenhaus committed:
Fix todo.remove_item frontend (#103108)

jbouwh committed:
Add strict typing for input_text (#103095)

CFenner committed:
Fix client id label in ViCare integration (#103111)

joostlek committed:
Abort config flow if Google Tasks API is not enabled (#103114)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

gjohansson-ST committed:
No aliases in workday (#103091)

CFenner committed:
Handle exception introduced with recent PyViCare update (#103110)

starkillerOG committed:
Bump reolink-aio to 0.7.12 (#103120)

Narmishka committed:
Move Freebox tests to fixtures (#103128)

joostlek committed:
Use right functions for fixtures in Freebox test (#103135)

bdraco committed:
Fix race in starting reauth flows (#103130)

bdraco committed:
Bump aiohomekit to 3.0.9 (#103123)

timmo001 committed:
Create update component for System Bridge (#102966)

* Create update component for System Bridge

* Add

piitaya committed:
Don't try to load resources in safe mode (#103122)

bdraco committed:
Avoid path construction for static files cache hit (#102882)

bdraco committed:
Allow non-admins to subscribe to the issue registry updated event (#103145)

emontnemery committed:
Improve reload of legacy groups (#102925)

* Improve reload of legacy groups

* Simplify reload

* Get rid of inner function

* Fix logic when there are no group.group entities

* Update homeassistant/components/group/__init__.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Fix type hints

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

rytilahti committed:
Bump python-kasa to 0.5.4 for tplink (#103038)

allenporter committed:
Improve fitbit oauth token error handling in config flow (#103131)

* Improve fitbit oauth token error handling in config flow

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update tests with updated error reason

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Bump aioesphomeapi to 18.2.1 (#103156)

jbouwh committed:
Fix mqtt is not reloading without yaml config (#103159)

bdraco committed:
Avoid enumerating the whole state machine on api service calls (#103147)

bramkragten committed:
Update frontend to 20231030.1 (#103163)

gjohansson-ST committed:
Bump python-holidays to 0.35 (#103092)

joostlek committed:
Bump aiowaqi to 3.0.0 (#103166)

jpbede committed:
Move base entity of system_bridge to own module (#103167)

* Move base entity of system_bridge to own module

* Add entity.py to .coveragerc

Xitee1 committed:
Add hardware version to Roomba (#103171)

mkmer committed:
Catch unexpected response in Honeywell (#103169)

catch unexpected response

tetele committed:
Support HassTurnOn/Off intents for lock domain (#93231)

* Support HassTurnOn/Off intents for lock domain

Fix https://github.com/home-assistant/intents/issues/1347

* Added tests

* Linting changes

* Linting

jimmyd-be committed:
Add reset filter counter button to Renson integration (#103126)

Add reset filter counter button

jpbede committed:
Use shorthand device info attribute for roomba (#103176)

jpbede committed:
Add MAC address to roborock device info (#103175)

bdraco committed:
Add bluetooth address to august (#103177)

jpbede committed:
Add MAC address to bsblan device info (#103180)

jpbede committed:
Use constant instead of plain key name for device info connections in roborock and roomba (#103182)

Xitee1 committed:
Fix roomba translation key mismatch (#103191)

Xitee1 committed:
Fix roomba error if battery stats are not available (#103196)

Lash-L committed:
Add button platform to Roborock (#103010)

* add button platform to roborock

* Update tests/components/roborock/test_button.py

Co-authored-by: Duco Sebel <74970928+DCSBL@users.noreply.github.com>

* Remove device class

* improve tests

* sort platforms

---------

Co-authored-by: Duco Sebel <74970928+DCSBL@users.noreply.github.com>

farmio committed:
Fix Fronius entity initialisation (#103211)

* Use None instead of raising ValueError if value invalid

* use async_dispatcher_send

allenporter committed:
Add modernized fitbit battery level sensor (#102500)

* Add modernized fitbit battery level sensor

* Use entity names for existing battery sensors

* Use icon from device class

* Update homeassistant/components/fitbit/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update tests with lower case naming

* Swap the names of the device battery sensors

* Revert "Swap the names of the device battery sensors"

This reverts commit c9516f6d06d77a27c5d7528cdbaa69153546bbbb.

* Update homeassistant/components/fitbit/sensor.py

Co-authored-by: Jan Vaníček <janvanicek@post.cz>

* Improve typing

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Jan Vaníček <janvanicek@post.cz>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

jbouwh committed:
Fix mqtt config validation error handling (#103210)

* Fix MQTT config check

* Fix handling invalid enity_category for sensors

* Improve docstr

* Update comment

* Use correct util for yaml dump

Shulyaka committed:
Shield service call from cancellation on REST API connection loss (#102657)

* Shield service call from cancellation on connection loss

* add test for timeout

* Apply suggestions from code review

* Apply suggestions from code review

* fix merge

* Apply suggestions from code review

joostlek committed:
Remove icon in Random (#103235)

joostlek committed:
Use shorthand attributes in Random (#103206)

elupus committed:
Remove measurement flag from timestamp in gardena bluetooth (#103245)

Remove measurement flag from timestamp

rappenze committed:
Fix fibaro event handling (#103199)

cgarwood committed:
Bump pyenphase to 1.14.1 (#103239)

Noltari committed:
Update aioairzone-cloud to v0.3.2 (#103258)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

bdraco committed:
Speed up websocket and ingress with aiohttp-zlib-ng (#103247)

bdraco committed:
Reduce overhead to run event triggers (#103172)

jpbede committed:
Use constant instead of plain key name for device info attributes (#103188)

* Use constant instead of plain key name for device info connections

* Some more device info constant changes

pjanuario committed:
add library logger info on ecoforest integration manifest (#103274)

abmantis committed:
Add buttons to connect/disconnect the Idasen Desk (#102433)

Co-authored-by: J. Nick Koston <nick@koston.org>

jbouwh committed:
Optmize timing excecutor timeout test (#103276)

eifinger committed:
Support multiple responses for service calls (#96370)

* add supports_response to platform entity services

* support multiple entities in entity_service_call

* support legacy response format for service calls

* revert changes to script/shell_command

* add back test for multiple responses for legacy service

* remove SupportsResponse.ONLY_LEGACY

* Apply suggestion

Co-authored-by: Allen Porter <allen.porter@gmail.com>

* test for entity_id remove None

* revert Apply suggestion

* return EntityServiceResponse from _handle_entity_call

* Use asyncio.gather

* EntityServiceResponse not Optional

* styling

---------

Co-authored-by: Allen Porter <allen.porter@gmail.com>

allenporter committed:
Add config flow to CalDAV (#103215)

* Initial caldav config flow with broken calendar platform

* Set up calendar entities

* Remove separate caldav entity

* Update tests after merge

* Readbility improvements

* Address lint issues

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Add checking for duplicate configuration entries

* Use verify SSL as input into caldav and simplify test setup

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Bump yalexs-ble to 2.3.2 (#103267)

gjohansson-ST committed:
Cleanup weather test (#103090)

* Cleanup weather test

* rename

* Clean off not used MockWeatherCompat

* conftest

* more cleanup

* Fin mod tests

* fix others

emontnemery committed:
Quote entity ids in entity excpetions (#103286)

bdraco committed:
Add debug logging for which adapter is used to connect bluetooth devices (#103264)

Log which adapter is used to connect bluetooth devices

This is a debug logging improvement to help users find problems
with their setup

Shutgun committed:
Add MAC adress to devolo Home Network DeviceInfo (#103290)

Add MAC adress to devolo Home Network devices

piitaya committed:
Add trigger selector for blueprint (#103050)

raman325 committed:
Fix firmware update failure (#103277)

marcelveldt committed:
Fix Matter 1.2 locks with specific unlatch/unbolt support (#103275)

tronikos committed:
Bump opower to 0.0.39 (#103292)

CoMPaTech committed:
Fix Plugwise Schedule selection (#103262)

cereal2nd committed:
Add loggers to the duotecno integration (#103300)

raman325 committed:
Fix zwave_js cover bug for Window Covering CC values (#103289)

* Fix cover bug for Window Covering CC values

* update test

* Fix fixture

* Remove no-op line from test

bdraco committed:
Remove useless inner function in the base Bluetooth coordinator (#103305)

Remove unless inner function in the base Bluetooth coordinator

ViViDboarder committed:
Bump py_nextbusnext to v1.0.2 to fix TypeError (#103214)

* Bump py_nextbusnext to v1.0.1 to fix TypeError

Currently throwing an error as a set is passed into the method that is currently
expecting a Sequence. That method is technically compatible with Iterable, so the
latest patch relaxes that restriction.

* Bump version to v1.0.2 to fix error message

starkillerOG committed:
Bump reolink-aio to 0.7.14 and improve typing of Reolink (#103129)

* Improve typing

* fix mypy

* Further improve typing

* Restore Literal typing

* Bump reolink_aio to 0.7.13

* Bump reolink-aio to 0.7.14

bdraco committed:
Bump SQLAlchemy to 2.0.23 (#103313)

jpbede committed:
Fix typo in Todoist config flow (#103317)

ludeeus committed:
Remove extra from traccar webhook (#103319)

jrieger committed:
Add translations to Workday state attributes (#103320)

ludeeus committed:
Bump pytraccar from 1.0.0 to 2.0.0 (#103318)

bdraco committed:
Cache the mime type of static files (#103281)

bdraco committed:
Small speed up to async_listen (#103307)

Avoid constructing an inner function each time we call async_listen
and use a partial which holds a reference to the function body with new
args instead of making another full function

emontnemery committed:
Prevent accidentally reusing an entity object (#102911)

* Prevent accidentally reusing an entity object

* Fix group reload service

* Revert "Fix group reload service"

* Improve test

* Add tests aserting entity can't be reused

jpbede committed:
Fix failing entity reuse test (#103342)

* Fix failing entity reuse test

* One more test

ViViDboarder committed:
Nextbus: Listify directions (#103337)

When a single value is returned, the list wrapper is not present in the
json payload. This patch ensures that the result is always a list.

engrbm87 committed:
Fix sensor unique id in Islamic prayer times (#103356)

bdraco committed:
Reduce template render overhead (#103343)

The contextmanager decorator creates a new context manager every
time its run, but since we only have a single context var, we can
use the same one every time. Creating the contextmanager was roughly
20% of the time time of the template render

I was a bit suprised to find it creates a new context manager
object every time https://stackoverflow.com/questions/34872535/why-contextmanager-is-slow

jrieger committed:
Report correct weather condition at night for Met (#103334)

* Report correct weather condition at night for Met, fixes #68369, fixes #89001

* Update homeassistant/components/met/weather.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

---------

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

farmio committed:
Use `setdefault()` in scaffold script for setting `hass.data` (#103338)

jrieger committed:
Add translations to DWD state attributes (#103359)

cgtobi committed:
Use pyatmo device type enum instead of string (#103030)

Kane610 committed:
Handle UniFi traffic rules not supported on older versions (#103346)

cgtobi committed:
Change log level to debug in Netatmo (#103365)

Change log level to debug

raman325 committed:
Add test for firmware update scenario (#103314)

jrieger committed:
Add translations to Tankerkoenig state attributes (#103363)

chrisx8 committed:
AirNow sensors should share device identifier (#103279)

* Entities from the same config entry should have the same device identifier
* Clean up unused device entries with no entities after sensor setup

Co-authored-by: James Pan <32176676+jzpan1@users.noreply.github.com>
Co-authored-by: Theo Ma <62950302+t1an-xyz@users.noreply.github.com>
Co-authored-by: Jonathan McDevitt <69861492+Jonmcd1@users.noreply.github.com>
Co-authored-by: Jadon Yack <86989502+jadonyack@users.noreply.github.com>
Co-authored-by: Ashton Foley <121987068+foleyash@users.noreply.github.com>
Co-authored-by: Leon Yan <138124222+leony7@users.noreply.github.com>

jpbede committed:
Add DeviceInfo to NINA (#103361)

jrieger committed:
Add translations to speedtest.net state attributes (#103362)

frenck committed:
Replace/restructure HomeWizard device fixtures to reflect reality (#103311)

jpbede committed:
Address late review for NINA (#103367)

Set device entry type

jpbede committed:
Set device entry type for dwd_weather_warnings (#103370)

* Set device entry type for dwd_weather_warnings

* Set integration type to service

jpbede committed:
Remove deprecated /config/server_control redirect (#103331)

frenck committed:
Set suggested display precision for HomeWizard Energy power sensors (#103369)

gjohansson-ST committed:
Remove platform YAML from Command line (#103202)

gjohansson-ST committed:
Remove counter configure service (#103204)

* Remove counter configure service after deprecation

* reproduce state

mkmer committed:
Address late review for blink (#103376)

* use self.coordinator

* Dont store coordinator

* revert unintended sensor change

* revert remove

* indention error

* revert

* Revert more

gjohansson-ST committed:
Remove platform yaml from myStrom (#103378)

dknowles2 committed:
Add diagnostics support to Schlage (#103347)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Remove unreachable code in logbook (#103309)

bdraco committed:
Bump protobuf to 4.25.0 (#103373)

gjohansson-ST committed:
Remove platform YAML from Qnap (#103377)

bieniu committed:
Don't assume that the `sleep` value is a dictionary in Tractive integration (#103138)

* Sleep value can be null

* Catch TypeError

gjohansson-ST committed:
Remove platform YAML from DWD Weather Warnings (#103379)

gjohansson-ST committed:
Remove platform YAML from LastFM (#103391)

slovdahl committed:
Bump vallox_websocket_api to 4.0.2 (#103339)

bdraco committed:
Pin jaraco.functools to fix builds and CI (#103406)

allenporter committed:
Fix CalDAV supported components check when configured from the UI (#103411)

* Fix CalDAV supported components check when configured from the UI

* Move async_get_calendars to a seprate file

* Simplify return value for async_get_calendars

emontnemery committed:
Differentiate between warnings and errors in check_config helper (#102902)

* Differentiate between warnings and errors in check_config helper

* Update tests

* Treat configuration errors in frontend and its dependencies as errors

* Improve test coverage

* Address review comments

* Improve test coverage

* Improve test coverage

* Address review comments

* Add comment

mib1185 committed:
Fix serial in Flo device information (#103427)

bdraco committed:
Bump bluetooth-data-tools to 0.14.0 (#103413)

changelog: https://github.com/Bluetooth-Devices/bluetooth-data-tools/compare/v1.13.0...v1.14.0

bdraco committed:
Bump zeroconf to 0.120.0 (#103412)

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.119.0...0.120.0

Noltari committed:
Update aioairzone-cloud to v0.3.5 (#103315)

* Update aioairzone-cloud to v0.3.3

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* Update aioairzone-cloud to v0.3.4

Reverts accidental TaskGroup changes.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* Update aioairzone-cloud to v0.3.5

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* Trigger Github CI

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
Co-authored-by: J. Nick Koston <nick@koston.org>

mkmer committed:
Use local variables in Blink (#103430)

local variables

mdz committed:
Fix litterrobot test failure due to time zone dependence (#103444)

* fix litterrobot test

* use a date in northern hemisehpere summer

janiversen committed:
Modbus set device_class in slaves (#103442)

cgtobi committed:
Bump pyatmo to v7.6.0 (#103410)

Signed-off-by: Tobias Sauerwein <cgtobi@gmail.com>

jbouwh committed:
Fix fritz entity category binary sensor is invalid (#103470)

gjohansson-ST committed:
Remove platform YAML from GeoJSON (#103393)

rappenze committed:
Refactor fibaro config flow test (#102604)

* Refactor fibaro config flow test

* Use constants from FlowResultType

* Extend tests with recovery after failure

* Add recovery from failure in all tests

Noltari committed:
Allow setting HVAC mode through set_temperature service in Airzone integration (#103185)

* airzone: climate: set_temperature: support ATTR_HVAC_MODE

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* tests: airzone: set_temp: check HVAC mode

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
Allow setting HVAC mode through set_temperature service in Airzone Cloud integration (#103184)

* airzone_cloud: climate: set_temperature: support ATTR_HVAC_MODE

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* tests: airzone_cloud: set_temp: check HVAC mode

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

engrbm87 committed:
Refactor tests for Islamic Prayer Times (#103439)

Refactor tests

mdz committed:
Handle smarttub sensor values being None (#103385)

* [smarttub] handle sensor values being None

* empty commit to rerun CI

* lint

* use const in test

* reorder checks

* use None instead of STATE_UNKNOWN

* empty commit to rerun CI

* check for STATE_UNKNOWN in test

* empty commit to rerun CI

janiversen committed:
modbus Allow swap: byte for datatype: string. (#103441)

joostlek committed:
Sort Withings sleep data on end date (#103454)

* Sort Withings sleep data on end date

* Sort Withings sleep data on end date

farmio committed:
Fix KNX expose default value when attribute is `None` (#103446)

Fix KNX expose default value when attribute is `null`

engrbm87 committed:
Store Islamic Prayer Times coordinator in 'entry_id' key (#103405)

allenporter committed:
Bump ical to 6.0.0 (#103482)

zxdavb committed:
Bump evohome-async to 0.4.4 (#103084)

* initial commit

* use correct attr

* fix hass-logger-period

* initial commit

* reduce footprint

* reduce footprint 2

* reduce footprint 3

* reduce footprint 4

* reduce footprint 6

* reduce footprint 7

* reduce footprint 8

* reduce footprint 9

* bump client to 0.4.1

* missing commit - changed method name

* bump client to 0.4.3

* bump client to 0.4.4

gjohansson-ST committed:
Add Check date service for Workday (#97280)

jbouwh committed:
Validate entity category for binary_sensor (#103464)

Noltari committed:
On Airzone cloud unload logout (#103487)

ludeeus committed:
Bump pycfdns from 2.0.1 to 3.0.0 (#103426)

frenck committed:
Update tailscale to 0.6.0 (#103409)

elupus committed:
Split out coordinator and add tests for nibe_heatpump (#103452)

* Separate coordinator in nibe heatpump

* Add tests for coordinator in nibe

* Correct errors in coordinator found during testing

* If coil is missing we should still write state
* async_shutdown did not call base class

* Add more tests for coordinator

* Add minimal test to climate

dupondje committed:
Use right equipment identifier in DSMR setup (#103494)

natekspencer committed:
Handle null data in WeatherFlow sensors (#103349)

Co-authored-by: J. Nick Koston <nick@koston.org>

zxdavb committed:
Remove redundant code from the evohome integration (#103508)

* remove unreachable except clause

* remove uneccesary try

jbouwh committed:
Add ServiceValidationError and translation support (#102592)

* Add ServiceValidationError

* Add translation support

* Extend translation support to HomeAssistantError

* Add translation support for ServiceNotFound exc

* Frontend translation & translation_key from caller

* Improve fallback message

* Set websocket_api as default translation_domain

* Add MQTT ServiceValidationError exception

* Follow up comments

* Revert removing gueard on translation_key

* Revert test changes to fix CI test

* Follow up comments

* Fix CI test

* Follow up

* Improve language

* Follow up comment

timmo001 committed:
Update systembridgeconnector to 3.9.4 (#103425)

* Update systembridgeconnector to 3.9.2

* Update systembridgeconnector to version 3.9.3

* Update systembridgeconnector to version 3.9.4

* Fix import

frenck committed:
Add device fixtures and tests for HomeWizard Energy Watersensor (#103383)

Co-authored-by: Duco Sebel <74970928+DCSBL@users.noreply.github.com>

frenck committed:
Update elgato to 5.1.0 (#103530)

bieniu committed:
Bump nextdns to version 2.0.1 (#103531)

bieniu committed:
Bump gios to version 3.2.1 (#103533)

bdraco committed:
Try to avoid re-parsing the content-type in hassio ingress if possible (#103477)

Co-authored-by: Stefan Agner <stefan@agner.ch>
Co-authored-by: Franck Nijhof <git@frenck.dev>

timmo001 committed:
Update systembridgeconnector version to 3.9.5 (#103515)

bieniu committed:
Bump accuweather to version 2.0.1 (#103532)

joostlek committed:
Bump nettigo-air-monitor to 2.2.1 (#103529)

synesthesiam committed:
Automatically convert TTS audio to MP3 on demand (#102814)

* Add ATTR_PREFERRED_FORMAT to TTS for auto-converting audio

* Move conversion into SpeechManager

* Handle None case for expected_extension

* Only use ATTR_AUDIO_OUTPUT

* Prefer MP3 in pipelines

* Automatically convert to mp3 on demand

* Add preferred audio format

* Break out preferred format

* Add ATTR_BLOCKING to allow async fetching

* Make a copy of supported options

* Fix MaryTTS tests

* Update ESPHome to use "wav" instead of "raw"

* Clean up tests, remove blocking

* Clean up rest of TTS tests

* Fix ESPHome tests

* More test coverage

jpbede committed:
Fix entry data typing in bsblan (#103544)

zxdavb committed:
Bump evohome-async to 0.4.6 (#103534)

* bump client to 0.4.5

* bump to 0.4.6

* adress lint mypy fails

frenck committed:
Update vehicle to 2.2.0 (#103545)

epenet committed:
Fix invalid MAC in samsungtv (#103512)

* Fix invalid MAC in samsungtv

* Also adjust __init__

timmo001 committed:
Add processes count sensor to System Bridge (#103516)

* Add processes count sensor to System Bridge

* Add processes string

starkillerOG committed:
Bump reolink_aio to 0.7.15 (#103548)

natekspencer committed:
Allow WeatherFlow devices to be removed (#103556)

Allow WeatherFlow devices to be removed if they haven't reported

cgarwood committed:
Bump pyenphase to 1.14.2 (#103553)

frenck committed:
Update pvo to 2.1.0 (#103551)

iMicknl committed:
Add diagnostics tests to Overkiz integration (#103560)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Noltari committed:
Update aioairzone-cloud to v0.3.6 (#103535)

engrbm87 committed:
Bump transmission-rpc to version 7.0.3 (#103502)

* Bump transmission-rpc to version 7.0.3

* Change `date_added` to `added_date`

allenporter committed:
Add read-only Caldav todo platform (#103415)

* Add Caldav todo enttiy for VTODO components

* Use new shared apis for todos

* Update homeassistant/components/caldav/todo.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update todo item conversion checks

* Iterate over results once

* Add 15 minute poll interval for caldav todo

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ffourcot committed:
Add hvac_action support to melcloud (#103372)

Since actions are shared between water tank (if any) and climate device,
hvac action can be idle even when heat pump is active

Co-authored-by: jan iversen <jancasacondor@gmail.com>

dupondje committed:
Update dsmr-parser to 1.3.1 to fix parsing issues (#103572)

bdraco committed:
Small cleanups to process_success_login (#103282)

gjohansson-ST committed:
Remove Ezviz detection sensitivity service (#103392)

bdraco committed:
Bump aioesphomeapi to 18.2.4 (#103552)

fetzerch committed:
Add select for partial position (garage door) in Overkiz (#99500)

egguy committed:
Add support for SomfyHeatingTemperatureInterface in Overkiz integration (#83514)

iMicknl committed:
Retrieve manufacturer and configuration_url from client in Overkiz integration (#103585)

Retrieve manufacturer and configuration_url from client

iMicknl committed:
Bump pyOverkiz to 1.13.0 (#103582)

bilak committed:
Add geofencing mode for Somfy Thermostat in Overkiz (#103160)

103136 Added geofencing mode for Overkiz thermostat

Lash-L committed:
Remove MyQ Integration (#103565)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

joostlek committed:
Parametrize DSMR serial config flow tests (#103524)

suaveolent committed:
fix: get_devices only checks for the first type (#103583)

rytilahti committed:
Bump python-songpal to 0.16 (#103561)

* Bump python-songpal to 0.16

* Set _attr_device_class

raman325 committed:
Add `zwave_js.refresh_notifications` service (#101370)

joostlek committed:
Raise exception when data can't be fetched in Opensky (#103596)

emontnemery committed:
Add file and line annotation to strings when loading yaml (#103586)

tetele committed:
Validate empty sentence triggers (#103579)

* Validate empty sentence triggers

* Add extra test for no sentences

* Remove extra line

---------

Co-authored-by: Michael Hansen <mike@rhasspy.org>

bdraco committed:
Ensure large payloads are compressed in the executor with aiohttp 3.9.0 (#103592)

synesthesiam committed:
Small code clean up (#103603)

dgomes committed:
Add V2C Trydan EVSE integration (#103478)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

jbouwh committed:
Fix metoffice test_forecast_subscription raises key error (#103598)

Allow test time change to be processed

joostlek committed:
Bump yt-dlp to 2023.10.13 (#103616)

frenck committed:
Update open-meteo to v0.3.1 (#103613)

emontnemery committed:
Correct line numbers in yaml node annotations (#103605)

allenporter committed:
Remove rainbird yaml config test fixtures (#103607)

emontnemery committed:
Fix yaml loader tests to test both C and Python implementations (#103606)

iMicknl committed:
Remove obstruction detected property for covers in Overkiz (#103597)

ludeeus committed:
Add instance id to the cloud integration (#103162)

frenck committed:
Update radios to v0.2.0 (#103614)

frenck committed:
Remove myself from verisure codeowners (#103625)

iMicknl committed:
Support continue_on_error for command execution in Overkiz (#103591)

jbouwh committed:
Fix entity category for sensor fails mqtt sensor platform setup (#103449)

mkmer committed:
Bump blinkpy to 0.22.3 (#103438)

engrbm87 committed:
Use EntityDescription for Transmission entities (#103581)

gladhorn committed:
Improve HomeKit description of what the PIN looks like (#103170)

Co-authored-by: J. Nick Koston <nick@koston.org>

emontnemery committed:
Rename YAML loader classes (#103609)

frenck committed:
Update sentry-sdk to 1.34.0 (#103623)

frenck committed:
Small cleanup in conftest mocks of PVOutput (#103628)

frenck committed:
Update syrupy to 4.6.0 (#103626)

frenck committed:
Update cryptography to 40.0.5 (#103624)

frenck committed:
Small cleanup in conftest mocks of Sensors.Community (#103630)

starkillerOG committed:
Add HDR switch to Reolink (#103550)

bdraco committed:
Bump dbus-fast to 2.13.1 (#103621)

* Bump dbus-fast to 2.13.0

changelog: https://github.com/Bluetooth-Devices/dbus-fast/compare/v2.12.0...v2.13.0

* no change re-release since upload failed due to running out of space on pypi

dgomes committed:
Add 4 new sensors to V2C (#103634)

* add 4 sensors

* no need for extra class

frenck committed:
Update pytest-picked to 0.5.0 (#103631)

ludeeus committed:
Bump awesomeversion from 23.8.0 to 23.11.0 (#103641)

dupondje committed:
Fix 5B Gas meter in dsmr (#103506)

* Fix 5B Gas meter in dsmr

In commit 1b73219 the gas meter broke for 5B.
As the change can't be reverted easily without removing the peak usage
sensors, we implement a workaround.

The first MBUS_METER_READING2 value will contain the gas meter data just
like the previous BELGIUM_5MIN_GAS_METER_READING did.
But this without the need to touch dsmr_parser (version).

Fixes: #103306, #103293

* Use parametrize

* Apply suggestions from code review

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

* Add additional tests + typo fix

---------

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>

dknowles2 committed:
Bump pydrawise to 2023.11.0 (#103638)

janiversen committed:
Reduce modbus validator by using table (#103488)

emontnemery committed:
Humanize core config errors in check_config helper (#103635)

janiversen committed:
Remove illegal int8 from modbus config (#103489)

dependabot[bot] committed:
Bump sigstore/cosign-installer from 3.1.2 to 3.2.0 (#103640)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vhartikainen committed:
Add vscode task to run changed tests (#103501)

iMicknl committed:
Remove redundant exception and catch NotSuchTokenException in Overkiz integration (#103584)

elupus committed:
Extend climate tests for nibe heatpump (#103522)

cereal2nd committed:
Add duotecno OFF hvac mode (#103223)

Lash-L committed:
Bump python-roborock to 0.36.0 (#103465)

emontnemery committed:
Allow removing an entity more than once (#102904)

frenck committed:
Update black to 23.11.0 (#103644)

hesselonline committed:
Wallbox Add Authentication Decorator (#102520)

jbouwh committed:
Fix entity category for binary_sensor fails setup (#103511)

ccrepin committed:
Change NP, NO code to Disarmed (#103617)

jpbede committed:
Move met coordinator to own module (#103546)

allenporter committed:
Add support for deleting To-do items in Google Tasks (#102967)

* Add support for deleting To-do items in Google Tasks

* Cleanup multipart test

* Fix comments

* Add additional error checking to increase coverage

* Apply suggestions and fix tests

nyroDev committed:
Merge remote-tracking branch 'refs/remotes/origin/overkiz/execute_multiple_commands' into overkiz/execute_multiple_commands

nyroDev commented:
Oops, my merge/rebase messed up the whole PR.
So sorry for all personns that have been notified for nothing.

I'm closing this PR and start a new one to not bother all of you...
